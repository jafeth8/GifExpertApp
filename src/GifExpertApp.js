import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GridGif } from './components/GridGif';

export const GifExpertApp = () => {
    //const categories=['One Punch','Samurai X','Dragon Ball'];
    
    
    const [categories, setcategories] = useState(['no game no life'])

    /*
    const handleAdd=()=>{
        //let nuevoValor=categories.push('Hunter x Hunter');
        //setcategories(nuevoValor)
        //categories.push('Hunter x Hunter')
        //console.log(categories)

        //setcategories([...categories,'Hunter x Hunter'])
        setcategories(cats=>[...cats,'Hunter x Hunter'])
    };
    */
    return (
        <>
          <h2>GifExpertApp</h2>
          <AddCategory setcategories={setcategories}/>
          <hr/>
          <ol>
              {
                categories.map(element=>(<GridGif key={element} category={element}/>))
              }
          </ol> 
        </>
    )
}
