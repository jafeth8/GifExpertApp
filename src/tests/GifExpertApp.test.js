import "@testing-library/jest-dom"
import {shallow} from "enzyme"
import { GifExpertApp } from "../GifExpertApp"

describe('pruebas en <GifExpertApp/>', () => {
    
    

    test('mostrar correctamente el componente', () => {
        const wrapper=shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('AddCategory').exists()).toBe(true);
        //expect(wrapper.find('GridGif').exists()).toBe(true);
    })

    test('debe mostrar una lista de categorias', () => {
        const categories=['One punch man', 'no game no life'];
        const wrapper=shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GridGif').length).toBe(categories.length);
    })
    

  
    
    
})
