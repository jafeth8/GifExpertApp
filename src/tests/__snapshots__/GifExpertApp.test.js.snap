// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp/> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol>
    <GridGif
      category="One punch man"
      key="One punch man"
    />
    <GridGif
      category="no game no life"
      key="no game no life"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en <GifExpertApp/> mostrar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
